public/app/plugins/datasource/influxdb/components/ConfigEditor.tsx

```diff
@@ -289,7 +267,60 @@ export class ConfigEditor extends PureComponent<Props> {
           </div>
         </div>
 
-        {options.jsonData.version === InfluxVersion.Flux ? this.renderInflux2x() : this.renderInflux1x()}
+        {options.jsonData.version === InfluxVersion.Flux && (
+          <InfoBox>
+            <h5>Support for flux in Grafana is currently in beta</h5>
+            <p>
+              Please report any issues to: <br />
+              <a href="https://github.com/grafana/grafana/issues/new/choose">
+                https://github.com/grafana/grafana/issues
+              </a>
+            </p>
+          </InfoBox>
+        )}
+
+        <DataSourceHttpSettings
+          showAccessOptions={true}
+          dataSourceConfig={options}
+          defaultUrl="http://localhost:8086"
+          onChange={onOptionsChange}
+        />
+
+        <div className="gf-form-group">
+          <div>
+            <h3 className="page-heading">InfluxDB Details</h3>
+          </div>
+          {options.jsonData.version === InfluxVersion.Flux ? this.renderInflux2x() : this.renderInflux1x()}
+          <div className="gf-form-inline">
+            <div className="gf-form">
+              <InlineFormLabel
+                tooltip="Max number of series that will be returned from the data source query. Defaults to 1000."
+                className="width-10"
+              >
+                Max series
+              </InlineFormLabel>
+              <div className="width-20">
+                <Input
+                  placeholder="1000"
+                  type="number"
+                  className="width-20"
+                  value={this.state.maxSeries}
+                  onChange={(event) => {
+                    // We duplicate this state so that we allow to write freely inside the input. We don't have
+                    // any influence over saving so this seems to be only way to do this.
+                    this.setState({ maxSeries: event.currentTarget.value });
+                    const val = parseInt(event.currentTarget.value, 10);
+                    updateDatasourcePluginJsonDataOption(
+                      this.props,
+                      'maxSeries',
+                      Number.isFinite(val) ? val : undefined
+                    );
+                  }}
+                />
+              </div>
+            </div>
+          </div>
+        </div>
```

Comment:
Just a nit: I think you can use `InlineField` here: https://developers.grafana.com/ui/latest/index.html?path=/story/forms-inlinefield--basic